File: closures/ex2.exs
outer = fn() ->
  outerVal = 10
  IO.puts "outer function created outerVal: #{outerVal}"
  fn() ->
    outerVal = outerVal + 1
    IO.puts "closure updated outerVal to: #{outerVal}"
  end
end

inner = outer.()
inner.()
inner.()
inner.()

File: closures/ex2.js
function outer() {
  var outerVal = 10;
  console.log(`outer function created outerVal: ${outerVal}`);
  return () => {
	  outerVal++;
    console.log(`closure updated outerVal to: ${outerVal}`);
  }
}

inner = outer();
inner();
inner();
inner();
